File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letter(string text);
int count_word(string text);
int count_sentences(string text);

int main(void)
{
    // Prompt the user for some text
    string Text = get_string("Text: ");

    // Count the number of letters, words, and sentences in the text
    int num_letter = count_letter(Text);
    int num_word = count_word(Text);
    int sentences = count_sentences(Text);

    // Compute the Coleman-Liau index
    float L = ((float) num_letter / (float) num_word) *
              100.0; // average number of letters per 100 words in the text
    float S = ((float) sentences / (float) num_word) *
               100.0; // the average number of sentences per 100 words in the text
    float index = 0.0588 * L - 0.296 * S - 15.8;

    // Print the grade level
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", (int) round(index));
    }
}

int count_letter(string text)
{
    int letters = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    return letters;
}

int count_word(string text)
{
    int words = 1;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == ' ')
        {
            words++;
        }
    }
    return words;
}

int count_sentences(string text)
{
    int sentence = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentence++;
        }
    }    return sentence;
}

File: sentimental-readability/readability.py
text = input("Text: ")

word = 0
letters = 0
sentences = 0

# Split the text into words based on spaces and count them
words = text.split()
word = len(words)

# Iterate through each character in the text
for char in text:
    if char.isalpha():
        letters += 1  # Count letters

    if char in [".", "!", "?"]:
        sentences += 1  # Count sentences based on sentence-ending punctuation

# Calculate the averages for L (letters per 100 words) and S (sentences per 100 words)
L = (float(letters) / float(word)) * 100.0
S = (float(sentences) / float(word)) * 100.0

# Compute the Coleman-Liau index
Coleman_Liau = 0.0588 * L - 0.296 * S - 15.8

# Round the Coleman-Liau index to the nearest whole number
grade = round(Coleman_Liau)

# Output the grade level
if grade < 1:
    print("Before Grade 1")
elif grade >= 16:
    print("Grade 16+")
else:
    print(f"Grade {grade}")
